### PR TITLE
Adds a cryostorage version of the ghost spawner.

### PR DESCRIPTION
**Pull request by Shadowlight213**
Has infinite uses and can be "reentered" (mob just deletes) by clicking on it with an empty hand.
Requested by @Bawhoppen for some sort of ghost bar thing.

### CODE REVIEW

**Comment by Archie700** [comment]
So is this for people who are pretty much taken completely out of the round?

**Comment by Bawhoppen** [comment]
@Archie700 theres already sleeper spawners in lavaland this is just an infinite use one for a specific purpose

**Comment by Archie700** [comment]
Yeah but I guess this pr is for when deadchat gets too quiet and there's nothing to do.

You might have to code it on centcomm or ctf z-level though

**Comment by Bawhoppen** [comment]
I have a plan for it already

File: code/modules/awaymissions/corpse.dm [hunk 1]
```diff
@@ -12,6 +12,8 @@
 	var/instant = FALSE	//fires on New
 	var/flavour_text = "The mapper forgot to set this!"
 	var/faction = null
+	var/permanent = FALSE	//Set to true for infinite mob spawn I guess.
```
Comment (Jordie0608) [inline code/modules/awaymissions/corpse.dm]: You could make this a var/uses instead and have -1 be infinite in case someone wants to do that sometime

Comment (Shadowlight213) [inline code/modules/awaymissions/corpse.dm]: I'll probably do both tbh. The permanent var will say if it disappears after it is used up, and the uses var would control the actual spawning.

**Comment by duncathan** [comment]
@Shadowlight213

**Comment by Shadowlight213** [comment]
This is merge ready.

File: code/modules/awaymissions/corpse.dm [hunk 2]
```diff
@@ -21,6 +24,9 @@
 /obj/effect/mob_spawn/attack_ghost(mob/user)
 	if(ticker.current_state != GAME_STATE_PLAYING || !loc)
 		return
+	if(uses == 0)
```
Comment (duncathan) [inline code/modules/awaymissions/corpse.dm]: `if(!uses)`

File: code/modules/awaymissions/corpse.dm [hunk 3]
```diff
@@ -77,7 +84,10 @@
 			for(var/objective in objectives)
 				MM.objectives += new/datum/objective(objective)
 		special(M)
-	qdel(src)
+	if(uses > 0)
+		uses--
+	if(!permanent && (uses == 0))
```
Comment (duncathan) [inline code/modules/awaymissions/corpse.dm]: `if(!permanent && !uses)`

**Pull request merged**